lib/dest/writeContents/index.js

```diff
@@ -1,11 +1,23 @@
 'use strict';
 
-var fs = require('fs');
+var fs = require('graceful-fs');
+var assign = require('object-assign');
+
 var writeDir = require('./writeDir');
 var writeStream = require('./writeStream');
 var writeBuffer = require('./writeBuffer');
 var writeSymbolicLink = require('./writeSymbolicLink');
 
+// TODO include sticky/setuid/setgid, i.e. 7777?
+var MASK_MODE = parseInt('0777', 8);
```

Comment:
@phated Not sure what you mean... If you meant dropping the zero, so it would leave `parseInt('777', 8)` -- that would do the exact same thing, since it is the same number...

I added the leading zero to emphasize we are now actively ignoring any sticky/setuid/setgid bits that users might have set on `stat.mode`. Notice that this does not set those bits on extant files on disk to 0 -- all this mask does is ignore any bits on `stat.mode` that are not rwx flags. And the TODO comment questioned if we wanted to allow users to set those other bits as well.

Either way, it is separate from the issue you linked, although if we were to allow these bits to be set by the user, and reflected on disk by `dest`, then there should be a test for it which doesn't use the `realMode` function of course (awkward description of what it does, acually).